core/src/main/java/tc/oc/pgm/menu/InventoryMenu.java

```diff
@@ -1,50 +1,75 @@
 package tc.oc.pgm.menu;
 
 import static com.google.common.base.Preconditions.checkArgument;
-import static com.google.common.base.Preconditions.checkNotNull;
 
+import com.google.common.collect.Lists;
+import java.util.List;
 import java.util.WeakHashMap;
 import javax.annotation.Nullable;
 import net.kyori.text.TranslatableComponent;
 import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.Listener;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.event.inventory.InventoryCloseEvent;
 import org.bukkit.inventory.Inventory;
 import org.bukkit.inventory.ItemStack;
+import tc.oc.pgm.api.PGM;
+import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchScope;
+import tc.oc.pgm.api.match.event.MatchUnloadEvent;
 import tc.oc.pgm.api.player.MatchPlayer;
 import tc.oc.pgm.observers.ObserverToolsMatchModule;
 import tc.oc.pgm.util.StringUtils;
 import tc.oc.pgm.util.text.TextTranslations;
 
-public abstract class InventoryMenu {
+public class InventoryMenu implements Listener {
 
   protected static final int ROW_WIDTH = 9; // Number of columns per row
   protected static final int MAX_ROWS = 6; // Max allowed row size
 
+  private final List<InventoryMenuItem> inventoryMenuItems;
+
   private final WeakHashMap<MatchPlayer, InventoryMenu> viewing =
       new WeakHashMap<>(); // Map of players who are viewing the gui, along with the menu
-  private final String title; // Title of the inventory
+  private final TranslatableComponent title; // Title of the inventory
```

Comment:
```suggestion
  private final Component title; // Title of the inventory
```
Probably best to make this a Component, a menu title in the future may not come from a translation. Such as if the menu name was that of a username.